client/question/services/FeedbackGeneratorService.js

```diff
@@ -417,21 +515,56 @@ tie.factory('FeedbackGeneratorService', [
        * @returns {Feedback}
        */
       getFeedback: function(tasks, codeEvalResult, rawCodeLineIndexes) {
+        // If the user receives the same error message increment the same error
+        // counter.
+        if (previousErrorString &&
+            previousErrorString === codeEvalResult.getErrorString()) {
+          _updateCounters(ERROR_COUNTER_SAME_RUNTIME);
+        }
```

Comment:
Just to check: is the intended behavior that, if the above condition does not hold, the counters etc. remain as they are (and don't get reset)? If so, that's fine. I just wanted to confirm.